src/pages/collab/index.tsx

```diff
@@ -0,0 +1,247 @@
+import { useEffect, useRef, useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { api } from "~/utils/api";
+
+import LoadingIcon from "../../components/collab/LoadingIcon";
+
+const MatchRequestPage = () => {
+  const [difficulty, setDifficulty] = useState(-1);
```

Comment:
I think this is too many hooks to read and not really needed when we can do something like the following
<https://blog.logrocket.com/using-react-usestate-object/>

Reply:
Refactored in new commit